tests/testthat/test-interface.R

```diff
@@ -184,3 +183,42 @@ test_that("get.edge.ids() deprecation", {
   expect_snapshot(get.edge.ids(g, 1:2))
   expect_snapshot(get.edge.ids(g, 1:2, multi = TRUE), error = TRUE)
 })
+
+test_that("get_edge_id() works with data frame", {
+  g <- make_full_graph(3, directed = FALSE)
+  el_df <- data.frame(from = c(1, 1), to = c(2, 3))
+  expect_equal(get_edge_ids(g, el_df), c(1, 2))
+})
+
+test_that("get_edge_id() works with matrices", {
+  g <- make_full_graph(10)
+  mat <- matrix(c(1, 2, 1, 3, 1, 4), 3, 2, byrow = TRUE)
+  expect_equal(get_edge_ids(g, mat), c(1, 2, 3))
+
+  mat <- matrix(c(1, 2), 1, 2)
+  expect_equal(get_edge_ids(g, mat), 1)
+})
+
+test_that("get_edge_id() errors correctly for wrong vp", {
+  g <- make_full_graph(3, directed = FALSE)
+  el_g <- make_empty_graph()
+  expect_snapshot(error = TRUE, {
+    get_edge_ids(g, el_g)
+  })
+  expect_error(get_edge_ids(g, NULL))
+  expect_error(get_edge_ids(g, NA))
+
+  V(g)$name <- letters[1:3]
+  df <- data.frame(from = c("a", "b"), to = c(1, 2))
+  expect_snapshot(error = TRUE, {
+    get_edge_ids(g, df)
+  })
+})
+
+test_that("get_edge_id() errors correctly for wrong matrices", {
+  g <- make_full_graph(10)
+  mat <- matrix(c(1, 2, 3, 4), 2, 2)
```

Comment:
```suggestion
  mat <- matrix(c(1, 2, 3, 4), nrow = 2, ncol = 2)
```

Reference: https://design.tidyverse.org/call-data-details.html